crates/rome_js_analyze/src/semantic_analyzers/nursery/no_confusing_arrow.rs

```diff
@@ -0,0 +1,113 @@
+use crate::{semantic_services::Semantic, JsRuleAction};
+use rome_analyze::{context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_diagnostics::Applicability;
+use rome_js_factory::make;
+use rome_js_syntax::{
+    AnyJsExpression, AnyJsFunctionBody, JsArrowFunctionExpression, JsSyntaxKind, JsSyntaxToken,
+};
+use rome_rowan::BatchMutationExt;
+
+declare_rule! {
+    /// Disallow arrow functions where they could be confused with comparisons.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-confusing-arrow
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var x = a => 1 ? 2 : 3;
+    /// var x = (a) => 1 ? 2 : 3;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var x = a => (1 ? 2 : 3);
+    /// var x = (a) => (1 ? 2 : 3);
+    /// var x = (a) => {
+    ///     return 1 ? 2 : 3;
+    /// };
+    /// var x = a => { return 1 ? 2 : 3; };
+    /// ```
+    ///
+    pub(crate) NoConfusingArrow {
+        version: "next",
+        name: "noConfusingArrow",
+        recommended: false,
+    }
+}
+
+impl Rule for NoConfusingArrow {
+    type Query = Semantic<JsArrowFunctionExpression>;
```

Comment:
```suggestion
    type Query = Ast<JsArrowFunctionExpression>;
```

We have several query type. Since you don't use the semantic model, `Ast` is enough. By the way the rule should be moved from `semantic_analyzer/` folder to `analyzer/` folder.